gridt/src/app/core/secure-storage.service.ts

```diff
@@ -23,23 +28,24 @@ export class SecureStorageService {
           }
         }
       ).catch( () => observer.error(`Key "${key}" does not exist in the secure storage.`) );
-    });
+    }).pipe(takeUntil(this.timer));
```

Comment:
Please leave a comment as to why this is necessary. (We know now, but for future reference.)